src/module.c

```diff
@@ -145,10 +145,36 @@ int AddDocumentCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
   LG_DEBUG("Adding doc %s with %d fields\n", RedisModule_StringPtrLen(doc.docKey, NULL),
            doc.numFields);
   RSAddDocumentCtx *aCtx = NewAddDocumentCtx(sp, &doc);
+
```

Comment:
See L139 - the document is apparently being overridden? You'd need to load the existing document in a 'new' document and then merge the fields somehow

Reply:
Why? The document is saved at this stage, reloading into the same object will just load the merged state from redis. Unless order is important or something (which it shouldn't be)

Reply:
Brain fart on semantics. I'm used to "another" DB's upsert semantics. :)